cmd/incus/top.go

```diff
@@ -0,0 +1,162 @@
+package main
+
+import (
+	"bytes"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"io"
+	"net/http"
+	"strings"
+
+	"github.com/spf13/cobra"
+
+	cli "github.com/lxc/incus/v6/internal/cmd"
+	"github.com/lxc/incus/v6/internal/i18n"
+	"github.com/lxc/incus/v6/internal/server/metrics"
+)
+
+type cmdTop struct {
+	global *cmdGlobal
+}
+
+// Command is a method of the cmdTop structure that returns a new cobra Command for managing command aliases.
+func (c *cmdTop) Command() *cobra.Command {
+	cmd := &cobra.Command{}
+	cmd.Use = usage("top")
+	cmd.Short = i18n.G("Manage command aliases")
+	cmd.Long = cli.FormatSection(i18n.G("Description"), i18n.G(
+		`Manage command aliases`))
+
+	// Workaround for subcommand usage errors. See: https://github.com/spf13/cobra/issues/706
+	cmd.Args = cobra.NoArgs
+	cmd.RunE = c.Run
+	return cmd
+}
+
+// Run is a method of the cmdTop structure. It implements the logic to call `incus top`.
+// This function implements the `top` command. It queries the metrics API at (/1.0/metrics) and renders a list of
+// instances with their CPU, memory and disk usage columns
+func (c *cmdTop) Run(cmd *cobra.Command, args []string) error {
+	conf := c.global.conf
+
+	remote, path, err := conf.ParseRemote("/1.0/metrics")
+	if err != nil {
+		return err
+	}
+
+	// Validate path
+	if !strings.HasPrefix(path, "/") {
+		return fmt.Errorf(i18n.G("Query path must start with /"))
+	}
+
+	d, err := conf.GetInstanceServer(remote)
+	if err != nil {
+		return err
+	}
+
+	// Perform the query
```

Comment:
So this part is odd as you're trying to directly hit `/1.0/metrics` to fetch the information when you could just run `d.GetMetrics()` to get the same data.

Reply:
Ah silly miss of mine- didn't realize d.GetMetrics existed. Will change this accordingly.